src/SAML2/Attribute/AttributeSet.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Surfnet\SamlBundle\SAML2\Attribute;
+
+use ArrayIterator;
+use Countable;
+use IteratorAggregate;
+use Serializable;
+
+class AttributeSet implements IteratorAggregate, Countable, Serializable
+{
+    /**
+     * @var Attribute[]
+     */
+    private $attributes = [];
+
+    /**
+     * @param Attribute $attribute
+     */
+    public function add(Attribute $attribute)
+    {
+        if (!$this->contains($attribute)) {
+            $this->attributes[] = $attribute;
+        }
+    }
+
+    /**
+     * @param Attribute $otherAttribute
+     * @return bool
+     */
+    public function contains(Attribute $otherAttribute)
+    {
+        foreach ($this->attributes as $attribute) {
+            if ($attribute->equals($otherAttribute)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @return ArrayIterator
+     */
+    public function getIterator()
```

Comment:
This docblock narrows the IteratorAggregate::getIterator() return type. Why not leave it out?